### ISSUE TITLE
Dedicated Salvager - Agent Checking

### ISSUE THREAD

**Issue by I3Questor** [ISeeDEDPpl/Questor#213]
I have a very strange issue where if the agent is not displayed in station services due to lag or wrong station or Questor starts in space after a disconnect or crash, then Questor just idles and does nothing.

Is there any way the Agent and blacklist checking can be disabled for the dedicated salvager?

I have tried commenting out these two sections but the bot just goes into Error.

10:58:17 [Settings] Loading Faction Blacklist
10:58:17 [Settings] Missions from the faction [Caldari State] will be declined
10:58:17 [Settings] Missions from the faction [Gallente Federation] will be declined
10:58:17 [Settings] Missions from the faction [Minmatar Republic] will be declined
10:58:17 [Settings] Missions from the faction [Amarr Empire] will be declined
10:58:17 [Settings] Missions from the faction [CONCORD Assembly] will be declined
10:58:17 [Settings] Missions from the faction [Ammatar Mandate] will be declined
10:58:17 [Settings] Missions from the faction [Khanid Kingdom] will be declined
10:58:17 [Settings] Missions from the faction [Thukker Tribe] will be declined
10:58:17 [Settings] Faction Blacklist now has [8] entries
10:58:17 [Cache] AgentId
10:58:17 [Cache.CurrentAgent] [ XXXXXXXXXX ] AgentID [ -1 ]
10:58:17 [Settings] Unable to locate agent [ XXXXXXXXXX ]
10:58:17 [Settings] Unable to locate agent [ XXXXXXXXXX ]
10:58:17 [Settings] Unable to locate agent [ XXXXXXXXXX ]
10:58:17 [Settings] Unable to locate agent [ XXXXXXXXXX ]
10:58:17 [Settings] Unable to locate agent [ XXXXXXXXXX ]
10:58:17 [Questor] Checking: Current time [10/15/2012 10:58:17] StopTimeSpecified [True] StopTime [ 16/10/2012 3:00:00 AM] ManualStopTime = 15/10/2012 8:56:07 PM
10:58:18 [Questor] Start Salvaging Behavior
10:58:47 [Questor] Running Innerspace command: windowtaskbar on _MyEVECharacterNameRedacted_
10:58:48 [Questor] Running Innerspace command: WindowText EVE - _MyEVECharacterNameRedacted_

**Comment by ISeeDEDPpl** [comment]
possibly, but it is going to be an undertaking to get that working properly... 
I'll take a look at it the next time I have a mountain of time to kill. 

btw, that log does not show the idling... can we get a log that does? the above log, unfortunately is normal(ish), the eve client sometimes does not have the agent info yet and thus returns null for a few seconds

**Comment by sigmastk** [comment]
I had similar issue with dedicated salvager, when started in space it couldn't find agentID.

I solved it with adding agent to contacts (ferox2552 was kind enough to share it)

There is post on q forums, log seems a lot similar (http://forum.thehackerwithin.com/yaf_postst917_Unable-to-find-agent-id.aspx)

**Comment by ISeeDEDPpl** [comment]
yeah, this needs to be resolved long term though because there are other behaviors that questor can do that do not involve an agent. 

In order to fix this right I am having to add a new setting for a "home" bookmark, which will be expected to be a station for now,but could in the future support other types of "home" bookmarks, like a safespot with cans or a POS corporate hangar (potentially). 

I should have something workable in a few days as we also need to re-work traveler a bit to support this.

**Comment by I3Questor** [comment]
Awesome mate, if you need testing on this, I am happy to oblige.  Just tell me what you need me to do.

I have one DS and make a second char into a DS if needed.

**Comment by Pluckyduck** [comment]
ISDP, i added the agent to the address book and have not had the relog problem as well.

**Comment by I3Questor** [comment]
Yeah I did the same and it seems to have sorted that issue.
